File: simter-id-data/src/main/java/tech/simter/id/dao/IdDao.java
package tech.simter.id.dao;

import reactor.core.publisher.Mono;

/**
 * The ID Dao.
 *
 * @author RJ
 */
public interface IdDao {
  /**
   * Get the next id for the specific type.
   *
   * @param t the type
   * @return the next id
   */
  Mono<Long> nextLong(String t);
}

File: simter-id-data/src/main/kotlin/tech/simter/id/dao/IdDao.kt
package tech.simter.id.dao

import reactor.core.publisher.Mono

/**
 * The ID Dao.
 *
 * @author RJ
 */
interface IdDao {
  /**
   * Get the next id for the specific type.
   *
   * @param[t] the type
   * @return the next id
   */
  fun nextLong(t: String): Mono<Long>
}